### PR TITLE
Fixed CivilianShipTest and massively upgraded ComputerAI

### PR DESCRIPTION
**Pull request by SilverMagnum**
Computer AI is almost completely brand new.
CivilianShipTest is 100% working and has 100% coverage of CivilianShip.
Added some destroyed ship lists into BattleshipModel, which should have been there since project 2. My bad.

### CODE REVIEW

File: src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java [hunk 1]
```diff
@@ -1,66 +1,235 @@
...
+        for(int i = 0; i < cpuMisses.size(); i++) {
+            if((row == cpuMisses.get(i).getRow()) && (col == cpuMisses.get(i).getCol())) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
+        //Currently this AI will loop from left-top to every other space.
```
Comment (ghost) [inline src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java:200]: Why are these lines commented out? Will we be needing this later?

**Comment by ghost** [comment]
Looks great. We will discuss the reasons why the lines were commented out later for future features.

File: src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java (hunk 1 above)
Comment (jlioy) [inline src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java:200]: Maybe we can remove some of these larger comment blocks before the final submission, so as to improve the readability of the code.

File: src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java [hunk 2]
```diff
@@ -1,66 +1,235 @@
...
+        }
+
+        adjCoords = cleanAdjacencyList(adjCoords, cpuModel.getComputermisses(), injuredShips);
+
+        //An optimization could be inserted here to shoot in the path as the rest of the hits.
+
+        return adjCoords;
+    }
+
+    private ArrayList<Point> cleanAdjacencyList(ArrayList<Point> dirtyAdjCoords, ArrayList<Point> misses, ArrayList<Point> hits) {
```
Comment (jlioy) [inline src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java:104]: These functions look like they are written in a clean fashion and appear to follow the coding style guidelines that were covered in class.

**Comment by jlioy** [comment]
These changes look to be appropriate for the features that we are supposed to be implementing.

**Pull request merged**